### ISSUE TITLE
Support unicode version of _<*>_ for idiom brackets

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#4502]
Currently idiom brackets only work if there is an operator called `_<*>_` in scope. However, the standard library uses the symbol `_⊛_` instead. I propose Agda should also support this symbol for the purpose of idiom brackets.

**Comment by L-TChen** [comment]
Yeah, I agree. On the other hand, I am also not very satisfied with how Agda chooses a specific symbol for me. IMO, the choice of symbol should be configurable, so something like
```
{-# BINDING ApplicativeApp   _⊛_  #-}
{-# BINDING ApplicativeUnit  pure #-}
```
would leave the choice to users instead of picking one for them. The module system also needs to take care of whether bindings are imported or not.

**Comment by jespercockx** [comment]
IIRC the problem is that the desugaring of `do` notation and idiom brackets is done very early in the parsing pipeline, at a point where we haven't yet processed all pragmas, so it is hard to make things configurable in this way.

**Comment by L-TChen** [comment]
Desugaring happens after the scope checking. Are pragmas processed later than that?

**Comment by jespercockx** [comment]
> Desugaring happens after the scope checking. Are pragmas processed later than that?

Yes, in https://github.com/agda/agda/blob/af74e45ed2177bed902c86910d4795ca90e20173/src/full/Agda/TypeChecking/Rules/Decl.hs#L705-L744 (there may also be other pragmas that are processed sooner, I'm not sure).

**Comment by andreasabel** [comment]
Pragmas are also processed in the scope checker:
https://github.com/agda/agda/blob/af74e45ed2177bed902c86910d4795ca90e20173/src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs#L2054-L2059
And even before that, in the nicifier:
https://github.com/agda/agda/blob/af74e45ed2177bed902c86910d4795ca90e20173/src/full/Agda/Syntax/Concrete/Definitions.hs#L1150-L1159
Somewhere there one can hook in the processing of the pragmas that define the names for resolving `do` and idiom brackets. 

P.S.: Things like fixity declarations are even processed out-of-order before scope-checking.
